### PR TITLE
Added firestoreClearData() testing sdk method

### PR DESCRIPTION
**Pull request by scottcrossen**
This PR exposes a new method for the @firebase/testing package to give users the ability to delete their entire database with one call. This is particularly useful in testing frameworks during the before/beforeEach lifecycle phases to isolate each test from one another. This only works with a properly updated Firestore Emulator (>v1.2.4)

### CODE REVIEW

File: packages/testing/index.ts [hunk 1]
```diff
@@ -29,5 +29,6 @@ export {
   initializeAdminApp,
   initializeTestApp,
   loadDatabaseRules,
-  loadFirestoreRules
+  loadFirestoreRules,
+  clearFirestoreData
```
Comment (ryanpbrewster) [inline packages/testing/index.ts]: sort alphabetically

File: packages/testing/src/api/index.ts [hunk 2]
```diff
@@ -220,6 +220,44 @@ export function loadFirestoreRules(
   });
 }
 
+export type clearFirestoreDataOptions = {
+  projectId: string;
+  databaseId?: string;
```
Comment (ryanpbrewster) [inline packages/testing/src/api/index.ts]: I would not implement a databaseId yet, these are not supported well

File: packages/testing/src/api/index.ts [hunk 3]
```diff
@@ -220,6 +220,44 @@ export function loadFirestoreRules(
...
+export function clearFirestoreData(
+  options: clearFirestoreDataOptions
+): Promise<void> {
+  if (!options.projectId) {
+    throw new Error('projectId not specified');
+  }
+
+  const optionsWithDefaults = Object.assign({}, options, {databaseId: '(default)'})
+
+  let client = new EMULATOR.FirestoreEmulator(
```
Comment (ryanpbrewster) [inline packages/testing/src/api/index.ts]: we should think about caching these stubs in the future

Comment (scottcrossen) [reply packages/testing/src/api/index.ts]: I agree!

File: packages/testing/src/api/index.ts [hunk 4]
```diff
@@ -220,6 +220,41 @@ export function loadFirestoreRules(
   });
 }
 
+export type clearFirestoreDataOptions = {
```
Comment (ryanpbrewster) [inline packages/testing/src/api/index.ts]: types should be PascalCase

Comment (scottcrossen) [reply packages/testing/src/api/index.ts]: [face-palm]

**Comment by scottcrossen** [comment]
I'll merge this once the emulator gets updated to the next release.

**Comment by tonymeng** [comment]
thoughts on mirroring this functionality for the rtdb?

**Comment by ryanpbrewster** [comment]
This functionality already exists for the RTDB, because RTDB deletes are recursive by default

**Comment by scottcrossen** [comment]
@tonymeng I'm not going to prioritize that because there's already a way as @ryanpbrewster mentioned. Additionally, we show them how in the [emulator quickstart](https://github.com/firebase/quickstart-nodejs/tree/master/database-emulator). If we get feedback from our developers that they'd like this testing-sdk method then I'd be happy to quickly write it out!

**Pull request merged**